backend/src/filter/validateOfTypeCondition.ts

```diff
@@ -0,0 +1,24 @@
+import { ArgumentError } from '../shared/errors';
+import { FilterCondition, OfTypeCondition } from '../shared/queries';
+import { checkValidOfTypeCondition } from './checkValidOfTypeCondition';
+import { formatErrorMessage } from './formatErrorMessage';
+
+export function validateOfTypeCondition(
+  condition: FilterCondition
+): OfTypeCondition {
+  const result = <OfTypeCondition>condition;
```

Comment:
Why is there an instant type change? So why is the argument type not of type `OfTypeCondition`?

Reply:
Thats because the function does in fact a protected cast. The check is in the next line.